examples/powertools-examples-core/kotlin/src/main/kotlin/helloworld/App.kt

```diff
@@ -0,0 +1,98 @@
+import software.amazon.lambda.powertools.tracing.CaptureMode
+import software.amazon.lambda.powertools.tracing.Tracing
+import software.amazon.lambda.powertools.tracing.TracingUtils
+import java.io.BufferedReader
+import java.io.IOException
+import java.io.InputStreamReader
+import java.net.URL
+import java.util.stream.Collectors
+
+/**
+ * Handler for requests to Lambda function.
+ */
+
+class App : RequestHandler<APIGatewayProxyRequestEvent?, APIGatewayProxyResponseEvent> {
+    @Logging(logEvent = true, samplingRate = 0.7)
+    @Tracing(captureMode = CaptureMode.RESPONSE_AND_ERROR)
+    @Metrics(namespace = "ServerlessAirline", service = "payment", captureColdStart = true)
+
+    override fun handleRequest(input: APIGatewayProxyRequestEvent?, context: Context?): APIGatewayProxyResponseEvent { // changed context to be optional so can pass null
+        val headers: MutableMap<String, String> = HashMap()
+        headers["Content-Type"] = "application/json"
+        headers["X-Custom-Header"] = "application/json"
+        MetricsUtils.metricsLogger().putMetric("CustomMetric1", 1.0, Unit.COUNT)
+        MetricsUtils.withSingleMetric("CustomMetrics2", 1.0, Unit.COUNT, "Another") { metric: MetricsLogger ->
+            metric.setDimensions(DimensionSet.of("AnotherService", "CustomService"))
+            metric.setDimensions(DimensionSet.of("AnotherService1", "CustomService1"))
+        }
+        LoggingUtils.appendKey("test", "willBeLogged")
+        val response = APIGatewayProxyResponseEvent().withHeaders(headers)
+        return try {
+            val pageContents = getPageContents("https://checkip.amazonaws.com")
+            log.info(pageContents)
+            TracingUtils.putAnnotation("Test", "New")
+            val output = """
+            {
+                "message": "hello world", 
+                "location": "$pageContents"
+            }
+            """.trimIndent()
+            TracingUtils.withSubsegment("loggingResponse") { subsegment: Subsegment? ->
+                val sampled = "log something out"
+                log.info(sampled)
+                log.info(output)
+            }
+            log.info("After output")
+            response.withStatusCode(200).withBody(output)
+        } catch (e: RuntimeException) {
+            response.withBody("{}").withStatusCode(500)
+        } catch (e: IOException) {
+            response.withBody("{}").withStatusCode(500)
+        }
+    }
+
+    @Tracing
+    private fun log() {
+        log.info("inside threaded logging for function")
+    }
+
+    @Tracing(namespace = "getPageContents", captureMode = CaptureMode.DISABLED)
+    @Throws(IOException::class)
+    private fun getPageContents(address: String): String {
+        val url = URL(address)
+        TracingUtils.putMetadata("getPageContents", address)
+        BufferedReader(InputStreamReader(url.openStream())).use { br -> return br.lines().collect(Collectors.joining(System.lineSeparator())) }
```

Comment:
`readText()` will do all the work for you, no need to join the lines
```suggestion
    return InputStreamReader(url.openStream()).use { reader ->
        reader.readText()
    }
```